Python 3.10 build action is failing on poetry dependency installation

## Bug Report
<!--
Hello, and thank you for reporting an issue!
Please fill out the points below, as it will make our process much easier.
-->

<!-- Please tell us about your environment -->
### Environment
  - `jrnl --diagnostic` output: <!-- Run the command and paste the output here -->
  - Install method: <!-- How did you install jrnl? (pipx, brew, etc) -->

### Current Behavior
<!--
Please put a short description of what is currently happening.
-->

In the "Install dependencies (Prerelease)" section, it's failing here:
```
Creating virtualenv jrnl in /home/runner/work/jrnl/jrnl/.venv
  Installing dependencies from lock file
  
    AttributeError
  
    'ProjectPackage' object has no attribute 'dev_requires'
  
    at /opt/hostedtoolcache/Python/3.10.0-rc.1/x64/lib/python3.10/site-packages/poetry/installation/installer.py:289 in _do_install
        285│ 
        286│         root = self._package
        287│         if not self.is_dev_mode():
        288│             root = root.clone()
      → 289│             del root.dev_requires[:]
        290│         elif self.is_dev_only():
        291│             root = root.clone()
        292│             del root.requires[:]
        293│ 
  Error: Process completed with exit code 1.
```

The failures are failing the whole job and it probably shouldn't, since Python 3.10 stable has not been released yet.

### Expected Behavior
<!--
Please write a short description of what you would expect to happen
(instead of what is currently happening).
-->

### Repro Steps
<!--
Please provide the steps to reproduce the problem. It is important for you to
be as precise as possible here, since more info will let us help you faster.
If applicable, please run jrnl with --debug and paste the output.
-->

### Other Information
<!-- (e.g. more detailed explanation, stacktraces, related
issues, suggestions how to fix, links for us to have context, eg.
stackoverflow, gitter, etc) -->

--- a/.github/workflows/testing.yaml
+++ b/.github/workflows/testing.yaml
@@ -64,7 +64,7 @@ jobs:
       if: ${{ matrix.python-version == '3.10-dev' }}
       run: |
         echo '::group::poetry'
-        pip install poetry==1.2.0a1
+        pip install poetry==1.2.0a2
         poetry config --local virtualenvs.in-project true
         echo '::endgroup::'
 

--- a/.github/workflows/testing.yaml
+++ b/.github/workflows/testing.yaml
@@ -9,6 +9,7 @@ on:
     - 'tests/**'
     - 'poetry.lock'
     - 'pyproject.toml'
+    - '.github/workflows/testing.yaml'
   pull_request:
     branches: [ develop ]
     paths:
@@ -17,6 +18,7 @@ on:
     - 'tests/**'
     - 'poetry.lock'
     - 'pyproject.toml'
+    - '.github/workflows/testing.yaml'
 
 jobs:
   test:

